src/classes/GE_PaymentServices.cls

```diff
@@ -0,0 +1,144 @@
+      from this software without specific prior written permission.
+
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2020
+* @group HGE
+* @group-content
+* @description Class holds payment services related methods.
+*/
+public with sharing class GE_PaymentServices {
+
+    String paymentMethodToken = null;
+
+    public GE_PaymentServices(String token) {
+        this.paymentMethodToken = token;
+    }
+
+    public HttpResponse makePurchaseCall() {
+        return makePurchaseCall(this.paymentMethodToken);
+    }
+
+    public HttpResponse makePurchaseCall(String token) {
+        // Create http request...
+        Http http = new Http();
+        HttpRequest request = new HttpRequest();
+        request.setEndpoint(getEndpointRoot() + getEndpoint(ElevateEndpoints.PURCHASE));
+        request.setMethod('POST');
+        request.setHeader('Content-Type', 'application/json;charset=UTF-8');
+        request.setHeader('jwttoken', getJwt());
+        request.setHeader('sfdo.merchantids', getMerchantIds());
+        request.setHeader('sfdo.username', getUsername());
+
+        // Set the body as a JSON object...
+        request.setBody('{"amount": ' + getAmount() + ',' +
+            '"currencyCode": "' + getCurrencyCode() + '",' +
+            '"paymentMethodToken": "' + token + '",' +
+            '"merchantId": "' + getMerchantId() + '",' +
+            '"gatewayId": "' + getGatewayId() + '",' +
+            '"email": "' + getEmail() + '",' +
+            '"firstName": "' + getFirstName() + '",' +
+            '"lastName": "' + getLastName() + '",' +
+            '"metadata": ' + getMetadata() + '}');
+
+        HttpResponse response = http.send(request);
+
+        return response;
+    }
+
+    private enum ElevateEndpoints { PURCHASE }
+
+    private String getEndpoint(ElevateEndpoints endpoint) {
+        if (endpoint === ElevateEndpoints.PURCHASE) {
+            return '/v1/payments/verified/purchase';
+        }
+        return null;
+    }
+
+    //
+    // Methods for retrieving request headers
+    //
+
+    private String getEndpointRoot() {
+        return 'https://7jpea5zk79.execute-api.us-west-2.amazonaws.com';
```

Comment:
Have we looked at/ruled out using named credentials to store this endpoint info?

Reply:
I don't know enough about packaging named credentials (reading up on it), but ideally I think we would like to use a named credential for this endpoint which would also solve the need to add this in remote site settings.

@dfuller114 - tagging you as I think you did the research on storing protected info that may have touched on named credentials. What do you think about storing this endpoint as a named credential?

Reply:
A named credential normally might be a good place for that. however, we're already going to likely use a protected custom object to store other information from payment services that they update when they register an org. it might make sense to just store the URL there also. Named Credential also makes the most sense when you want the named credential to handle the authentication. to just store the URL there without using the auth protocols built into it is underutilizing that feature a bit I think.

Storing it in the custom object will also make it easier to know where all the payment related protect info is stored. It would also make it easier for PS to update the URL if it changes instead of that responsibility being on us since they will already have code to update that object.